pkg/server/server_test.go

```diff
@@ -79,6 +82,45 @@ var _ = Describe("ApplicationsServer", func() {
 
 		Expect(res.Application.Name).To(Equal(name))
 	})
+	It("Authorize", func() {
+		ctx := context.Background()
+		provider := "github"
+		token := "token"
+
+		expectedToken, err := auth.Generate(auth.SecretKey, auth.ExpirationTime, provider, token)
+		Expect(err).NotTo(HaveOccurred())
+
+		res, err := appsClient.Authenticate(ctx, &pb.AuthenticateRequest{
+			ProviderName: provider,
+			AccessToken:  token,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(res.Token).To(Equal(expectedToken))
+	})
+	It("Authorize fails on wrong provider", func() {
+		ctx := context.Background()
+		provider := "wrong provider"
+		token := "token"
+
+		_, err := appsClient.Authenticate(ctx, &pb.AuthenticateRequest{
```

Comment:
We should be testing for a specific HTTP error code here

Reply:
not sure I understand, this is one of the possible errors `rpc error: code = Unknown desc = unknown provider name wrongprovider, expecting github or gitlab`

Reply:
Sorry I thought we had a reference to HTTP errors already. 

You can use the runtime package to return a specific error code. An error case might look like this:
```
	if err != nil {
		return nil, &runtime.HTTPStatusError{HTTPStatus: http.StatusUnauthorized, Err: err}
	}
```

We need to ensure the correct error code is being returned. The UI and API users will configure their behavior based on this code.